websocket_push_session.go

```diff
@@ -164,6 +172,44 @@ func (s *wsPushSession) Parameter(key string) string {
 	return s.parameters.Get(key)
 }
 
+func (s *wsPushSession) inErrorState() bool {
+	s.errorStateLock.RLock()
+	defer s.errorStateLock.RUnlock()
+
+	return s.errorStateActive
+}
+
+func (s *wsPushSession) setErrorState(on bool) {
+	s.errorStateLock.Lock()
+	defer s.errorStateLock.Unlock()
+
+	s.errorStateActive = on
+}
+
+func (s *wsPushSession) handlesErrorEvents() bool {
+	_, ok := s.parameters[enableErrorsQueryParam]
+	return ok
+}
+
+func (s *wsPushSession) sendWSError(ee elemental.Error) error {
+
+	s.setErrorState(true)
+	msgpack, json, err := prepareEventData(elemental.NewErrorEvent(ee, s.encodingWrite))
+	if err != nil {
+		s.close(websocket.CloseInternalServerErr)
```

Comment:
this is weird to close here, and log the error in the return.
You should log in here, then you don't even need to return an error

Reply:
Done ✅